### PR TITLE
Add IPv6 support to the nettest image

### LINKED ISSUE #70248: IPv6-only cluster e2e Testing Issues
**Which test(s) are failing**:

### Test regex is looking for an IPv4 address

[Fail] [sig-api-machinery] Downward API [It] should provide pod name, namespace and IP address as env vars [NodeConformance] [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2325

[Fail] [sig-api-machinery] Downward API [It] should provide host IP as an env var [NodeConformance] [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2325

---

### Container(s) used in the test can't do liveness/readiness probes using an IPv6 address.

[Fail] [sig-cli] Kubectl client [k8s.io] Guestbook application [It] should create and stop a working application  [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl/kubectl.go:1914

[Fail] [sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] [It] should perform canary updates and phased rolling updates of template modifications [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/statefulset_utils.go:323

[Fail] [k8s.io] [sig-node] PreStop [It] should call prestop when killing a pod  [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/node/pre_stop.go:158

[Fail] [k8s.io] Probing container [It] should *not* be restarted with a /healthz http liveness probe [NodeConformance] [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/common/container_probe.go:445

[Fail] [sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] [It] should perform rolling updates and roll backs of template modifications [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/statefulset_utils.go:323

[Fail] [sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] [It] Scaling should happen in predictable order and halt if any stateful pod is unhealthy [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/statefulset_utils.go:323

[Fail] [sig-apps] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] [It] Burst scaling should run to completion even with unhealthy pods [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/statefulset_utils.go:323

---

### Test is looking for an A record, vs an AAAA record:

[Fail] [sig-network] DNS [It] should provide DNS for the cluster  [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/dns_common.go:497

[Fail] [sig-network] DNS [It] should provide DNS for services  [Conformance] 
/workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/network/dns_common.go:497

**Since when has it been failing**: 

**Testgrid link**:

**Reason for failure**: (see above)

**Anything else we need to know**:
Can be reproduced with an IPv6 Kubernetes cluster (v1.11) and a Sonobuoy e2e run. The IPv6 cluster can be run locally with DinD:

```
wget https://cdn.rawgit.com/kubernetes-sigs/kubeadm-dind-cluster/master/fixed/dind-cluster-v1.11.sh
$ chmod +x dind-cluster-v1.11.sh
$ export IP_MODE=ipv6
$ ./dind-cluster-v1.11.sh up
```

/assign @timothysc 

/kind failing-test

### PR DESCRIPTION
**Pull request by aojea**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
/kind failing-test

> /kind feature
> /kind flake

**What this PR does / why we need it**:

The test [sig-node] PreStop should call prestop when killing a pod
[Conformance] use the nettest image for testing, but it turns out
that this image is configured to listen in the address 0.0.0.0.

Removing the address from the function http.ListenAndServe makes
it start listening on both IPv4 and IPv6 addresses.

**Which issue(s) this PR fixes**:

Fixes Partially #70248

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by ixdy** [comment]
this probably warrants a VERSION bump too?

**Comment by aojea** [comment]
@ixdy @listx I've bumped the version and rebased the image to distroless from scratch

File: test/images/nettest/nettest.go [hunk 1]
```diff
@@ -210,7 +210,7 @@ func main() {
 	http.HandleFunc("/write", state.serveWrite)
 	http.HandleFunc("/status", state.serveStatus)
 
-	go log.Fatal(http.ListenAndServe(fmt.Sprintf("0.0.0.0:%d", *port), nil))
+	go log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", *port), nil))
```
Comment (listx) [inline test/images/nettest/nettest.go]: Can you explain why `0.0.0.0` was dropped?

Comment (aojea) [reply test/images/nettest/nettest.go]: Dropping it  makes the server start listening in ipv4 and ipv6 ( if available). The ListenAndServe method uses  https://golang.org/src/net/dial.go?s=20989:21043#L677
> For TCP networks, if the host in the address parameter is empty or a literal unspecified IP address, Listen listens on all available unicast and anycast IP addresses of the local system

**Comment by aojea** [comment]
@listx @ixdy is this ok to test? 😄

**Pull request merged**